server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -0,0 +1,201 @@
+    private ConsumerResource consumerResource;
+
+    public static final String CREATE = "create";
+    public static final String DATA = "data";
+    public static final String PRINCIPAL = "principal";
+    protected static String prefix = "hypervisor_update_";
+
+    @Inject
+    public HypervisorUpdateJob(OwnerCurator ownerCurator, ConsumerCurator consumerCurator,
+            ConsumerResource consumerResource) {
+        this.ownerCurator = ownerCurator;
+        this.consumerCurator = consumerCurator;
+        this.consumerResource = consumerResource;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Executes {@link ConsumerResource#create(org.candlepin.model.Consumer, org.candlepin.auth.Principal,
+     *  java.utl.String, java.utl.String, java.utl.String)}
+     * Executes (@link ConusmerResource#updateConsumer(java.utl.String, org.candlepin.model.Consumer)}
+     * as a pinsetter job.
+     *
+     * @param context the job's execution context
+     */
+    public void toExecute(JobExecutionContext context) throws JobExecutionException {
+        try {
+            JobDataMap map = context.getMergedJobDataMap();
+            String ownerKey = map.getString(JobStatus.TARGET_ID);
+            Boolean create = map.getBoolean(CREATE);
+            Principal principal = (Principal) map.get(PRINCIPAL);
+
+            Owner owner = ownerCurator.lookupByKey(ownerKey);
+            if (owner == null) {
+                context.setResult("Nothing to do. Owner does not exist");
+                return;
+            }
+            byte[] data = (byte[]) map.get(DATA);
+            String json = decompress(data);
+            HypervisorList hypervisors = (HypervisorList) Util.fromJson(json, HypervisorList.class);
+            log.info("Hypervisor consumers for create/update: " + hypervisors.getHypervisors().size());
+
+            for (Consumer hypervisor : hypervisors.getHypervisors()) {
+                if (hypervisor.getHypervisorId() != null &&
+                        hypervisor.getHypervisorId().getHypervisorId() != null) {
+                    String hypervisorId = hypervisor.getHypervisorId().getHypervisorId();
+                    hypervisor.setType(new ConsumerType(ConsumerTypeEnum.HYPERVISOR));
+                    Consumer knownHost = consumerCurator.getHypervisor(hypervisorId, owner);
+                    if (knownHost == null) {
+                        if (!create) {
+                            throw new Exception("Unable to find hypervisor with id " +
+                                                hypervisorId + " in org " + ownerKey);
+                        }
+                        log.info("Registering new host consumer for hypervisor ID: {}", hypervisorId);
+                        consumerResource.create(hypervisor, principal, null, owner.getKey(), null);
+                    }
+                    else {
+                        consumerResource.updateConsumer(knownHost.getUuid(), hypervisor);
```

Comment:
A little worried here we're missing functional parity with the old API call, but if we do the above proposal for rspec (running the hypervisor resource spec against both API calls to update), it should tell us.